tests/sstable_mutation_test.cc

```diff
@@ -817,6 +817,60 @@ SEASTAR_THREAD_TEST_CASE(tombstone_in_tombstone2) {
     }).get();
 }
 
+// Reproducer for #4783
+static schema_ptr buffer_overflow_schema() {
+    static const auto s = [] {
+        schema_builder builder(make_lw_shared(schema(generate_legacy_id("test_ks", "test_tab"), "test_ks", "test_tab",
```

Comment:
@haaawk make_lw_shared is called by this static lambda.
Could this cause the error we saw:
```
sstable_mutation_test_g: /local/home/bhalevy/dev/scylla/seastar/include/seastar/core/shared_ptr_debug_helper.hh:63: void seastar::debug_shared_ptr_counter_type::check() const: Assertion `_cpu == std::this_thread::get_id()' failed.
```

Reply:
seems that it might be it. Thanks @bhalevy